worlds/celeste_open_world/__init__.py

```diff
@@ -0,0 +1,351 @@
+                checkpoint_loc.place_locked_item(self.create_item(item_name))
+                location_count -= 1
+
+        # Keys
+        if self.options.keysanity:
+            item_pool += [self.create_item(item_name) for item_name in self.active_key_names]
+        else:
+            for item_name in self.active_key_names:
+                key_loc: Location = self.multiworld.get_location(item_name, self.player)
+                key_loc.place_locked_item(self.create_item(item_name))
+                location_count -= 1
+
+        # Summit Gems
+        if self.options.gemsanity:
+            item_pool += [self.create_item(item_name) for item_name in self.active_gem_names]
+        else:
+            for item_name in self.active_gem_names:
+                gem_loc: Location = self.multiworld.get_location(item_name, self.player)
+                gem_loc.place_locked_item(self.create_item(item_name))
+                location_count -= 1
+
+        # Clutter Events
+        for item_name in self.active_clutter_names:
+            clutter_loc: Location = self.multiworld.get_location(item_name, self.player)
+            clutter_loc.place_locked_item(self.create_item(item_name))
+            location_count -= 1
+
+        # Interactables
+        item_pool += [self.create_item(item_name) for item_name in sorted(self.active_items)]
+
+        # Strawberries
+        real_total_strawberries: int = min(self.options.total_strawberries.value, location_count - goal_area_location_count - len(item_pool))
+        self.strawberries_required = int(real_total_strawberries * (self.options.strawberries_required_percentage / 100))
+
+        menu_region = self.multiworld.get_region("Menu", self.player)
+        if getattr(self, "goal_start_region", None):
+            menu_region.add_exits([self.goal_start_region], {self.goal_start_region: lambda state: state.has(ItemName.strawberry, self.player, self.strawberries_required)})
+        if getattr(self, "goal_checkpoint_names", None):
+            for region_name, location_name in self.goal_checkpoint_names.items():
+                checkpoint_rule = lambda state, location_name=location_name: state.has(location_name, self.player) and state.has(ItemName.strawberry, self.player, self.strawberries_required)
+                menu_region.add_exits([region_name], {region_name: checkpoint_rule})
+
+        menu_region.add_exits([self.epilogue_start_region], {self.epilogue_start_region: lambda state: (state.has(ItemName.strawberry, self.player, self.strawberries_required) and state.has(ItemName.house_keys, self.player))})
+
+        item_pool += [self.create_item(ItemName.strawberry) for _ in range(self.strawberries_required)]
+
+        # Filler and Traps
+        non_required_strawberries = (real_total_strawberries - self.strawberries_required)
+        replacement_filler_count = math.floor(non_required_strawberries * (self.options.junk_fill_percentage.value / 100.0))
```

Comment:
if you're not expecting the float to be negative int() and math.floor() should be equivalent

Reply:
I'd personally prefer it stylistically if you don't need flooring specifically. Or if you like it better you should probably change the line above using `int` instead so it's consistent. (Also why parens on the line above?)
Also you could just use floored division instead if you don't mind the division not directly being on the percentage.
```suggestion
        replacement_filler_count = (non_required_strawberries * self.options.junk_fill_percentage.value) // 100
```